packages/common/src/sites/_internal/applyDefaultCollectionMigration.ts

```diff
@@ -0,0 +1,105 @@
+import { getProp } from "../../objects/get-prop";
+import { IHubCollectionPersistance } from "../../search/types/IHubCatalog";
+import { WellKnownCollection } from "../../search/wellKnownCatalog";
+import { IModel } from "../../types";
+
+/**
+ * Adds default collections to site models that have the new catalog structure.
+ * These default collections will have the same name
+ *
+ * This migration simplifies display logic, as consuming components no longer
+ * have to merge `catalog.collections` with the default collection definitions.
+ *
+ * This migration also lays the foundation for customization, since editors will
+ * be able to modify these persisted default objects. For example, editors can
+ * change the default labels, display order, or mark a default as "hidden" so the
+ * public can't see it.
+ *
+ * @param model site model to migrate
+ * @returns a migrated model with default `IHubCollectionPersistance` objects added
+ * to the catalog
+ */
+export function applyDefaultCollectionMigration(model: IModel): IModel {
+  if (getProp(model, "item.properties.schemaVersion") >= 1.8) {
+    return model;
+  } else {
+    const baseCollectionKeys: WellKnownCollection[] = [
+      // TODO: add 'all' as a wellknown collection and figure out the
+      // ramifications of doing so across the app
+      "all" as any,
+      "dataset",
+      "document",
+      "site",
+      "appAndMap",
+      "document",
+    ];
+    const baseCollectionMap = baseCollectionKeys.reduce((map, key) => {
+      map[key] = {
+        // We chose to leave the label as "null" for a couple of reasons. First off,
+        // the default collection names are supposed to be translated and we don't
+        // have access to i18n services here. Second, because we temporarily have to
+        // serialize changes from catalog.collections back to site.data.values.searchCategories
+        // for backwards compatibility, we need a way to indicate whether a collection's
+        // name has been explicitly set or if it is relying on the default translation.
+        // As such, we only set the label on the collection object IFF an explicit
+        // override text has been configured.
+        label: null,
+        key,
+        include: [],
+        targetEntity: "item",
+        scope: {
+          targetEntity: "item",
+          collection: key,
```

Comment:
This is a weird thing that happened over the evolution of the `hubSearch` subsystem. `IHubCollection` has an `IQuery` scope which is passed to `hubSearch`. In an ideal world, `hubSearch` shouldn't even know that `IHubCollection` exists. However, by using this `collection: WellKnownCollection` field in `IQuery`, we're creating a circular "dependency" of sorts where `IHubCollection` knows about `IQuery`, and `IQuery` and `hubSearch` know about `IHubCollection`.

Before we merge this in, we'll want to evaluate whether `IQuery.collection` is the best semantic name for the field and make any needed changes.